Failure of "indices.freeze/10_basic/Cannot freeze write index for data stream" test case

**Build scan**:


Build Scans | https://gradle-enterprise.elastic.co/s/5u7b2zyqx7qaw

**Repro line**:

REPRODUCE WITH: ./gradlew ':x-pack:plugin:integTestRunner' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT" \
  -Dtests.method="test {p0=indices.freeze/10_basic/Cannot freeze write index for data stream}" \
  -Dtests.seed=8485F72911B39C9F \
  -Dtests.security.manager=true \
  -Dtests.locale=ro \
  -Dtests.timezone=BST \
  -Druntime.java=11 \
  -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*

REPRODUCE WITH: ./gradlew ':x-pack:plugin:integTestRunner' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT" \
  -Dtests.method="test {p0=indices.freeze/10_basic/Cannot freeze write index for data stream}" \
  -Dtests.seed=8485F72911B39C9F \
  -Dtests.security.manager=true \
  -Dtests.locale=ro \
  -Dtests.timezone=BST \
  -Druntime.java=11 \
  -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*

**Reproduces locally?**:

in progress...

**Applicable branches**:

master

**Failure history**:

no history

**Failure excerpt**:

```
16:03:22 org.elasticsearch.xpack.test.rest.XPackRestIT > test {p0=indices.freeze/10_basic/Cannot freeze write index for data stream} FAILED
16:03:22     java.lang.AssertionError: Failure at [indices.freeze/10_basic:133]: expected [2xx] status code but api [indices.put_index_template] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"composable template [my-template1] template after composition is invalid","stack_trace":"org.elasticsearch.ElasticsearchException$1: composable template [my-template1] template after composition is invalid\n\tat org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:644)\n\tat org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:572)\n\tat org.elasticsearch.rest.BytesRestResponse.build(BytesRestResponse.java:149)\n\tat org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:110)\n\tat org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:93)\n\tat org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:58)\n\tat org.elasticsearch.client.node.NodeClient.lambda$executeLocally$1(NodeClient.java:91)\n\tat org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:184)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50)\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:163)\n\tat org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:94)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$4.onFailure(MetadataIndexTemplateService.java:378)\n\tat org.elasticsearch.cluster.service.MasterService$SafeClusterStateTaskListener.onFailure(MasterService.java:508)\n\tat org.elasticsearch.cluster.service.MasterService$TaskOutputs.notifyFailedTasks(MasterService.java:441)\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:215)\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:647)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: java.lang.IllegalArgumentException: composable template [my-template1] template after composition is invalid\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:496)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$4.execute(MetadataIndexTemplateService.java:383)\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)\n\t... 9 more\nCaused by: java.lang.IllegalArgumentException: invalid composite mappings for [my-template1]\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1057)\n\tat org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:619)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1030)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:492)\n\t... 14 more\nCaused by: java.lang.IllegalArgumentException: expected timestamp field [@timestamp], but found no timestamp field\n\tat org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.validateTimestampFieldMapping(MetadataCreateDataStreamService.java:171)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1054)\n\t... 17 more\n"}],"type":"illegal_argument_exception","reason":"composable template [my-template1] template after composition is invalid","caused_by":{"type":"illegal_argument_exception","reason":"invalid composite mappings for [my-template1]","caused_by":{"type":"illegal_argument_exception","reason":"expected timestamp field [@timestamp], but found no timestamp field","stack_trace":"java.lang.IllegalArgumentException: expected timestamp field [@timestamp], but found no timestamp field\n\tat org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.validateTimestampFieldMapping(MetadataCreateDataStreamService.java:171)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1054)\n\tat org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:619)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1030)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:492)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$4.execute(MetadataIndexTemplateService.java:383)\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:647)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n"},"stack_trace":"java.lang.IllegalArgumentException: invalid composite mappings for [my-template1]\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1057)\n\tat org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:619)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1030)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:492)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$4.execute(MetadataIndexTemplateService.java:383)\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:647)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: java.lang.IllegalArgumentException: expected timestamp field [@timestamp], but found no timestamp field\n\tat org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.validateTimestampFieldMapping(MetadataCreateDataStreamService.java:171)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1054)\n\t... 17 more\n"},"stack_trace":"java.lang.IllegalArgumentException: composable template [my-template1] template after composition is invalid\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:496)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$4.execute(MetadataIndexTemplateService.java:383)\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697)\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319)\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214)\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:647)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: java.lang.IllegalArgumentException: invalid composite mappings for [my-template1]\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1057)\n\tat org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:619)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1030)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:492)\n\t... 14 more\nCaused by: java.lang.IllegalArgumentException: expected timestamp field [@timestamp], but found no timestamp field\n\tat org.elasticsearch.cluster.metadata.MetadataCreateDataStreamService.validateTimestampFieldMapping(MetadataCreateDataStreamService.java:171)\n\tat org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$19(MetadataIndexTemplateService.java:1054)\n\t... 17 more\n"},"status":400}]
```

--- a/x-pack/plugin/src/test/resources/rest-api-spec/test/indices.freeze/10_basic.yml
+++ b/x-pack/plugin/src/test/resources/rest-api-spec/test/indices.freeze/10_basic.yml
@@ -137,6 +137,11 @@
         name: my-template1
         body:
           index_patterns: [simple-data-stream1]
+          template:
+            mappings:
+              properties:
+                '@timestamp':
+                  type: date
           data_stream:
             timestamp_field: '@timestamp'
 

